server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/service/ClearMailRepositoryTask.java

```diff
@@ -57,20 +58,20 @@ public long getInitialCount() {
         }
     }
 
-    private final MailRepository mailRepository;
+    private final List<MailRepository> mailRepositories;
     private final AdditionalInformation additionalInformation;
 
-    public ClearMailRepositoryTask(MailRepository mailRepository, MailRepositoryUrl url) {
-        this.mailRepository = mailRepository;
-        this.additionalInformation = new AdditionalInformation(url, this::getRemainingSize);
+    public ClearMailRepositoryTask(List<MailRepository> mailRepositories, MailRepositoryPath path) {
+        this.mailRepositories = mailRepositories;
+        this.additionalInformation = new AdditionalInformation(path, this::getRemainingSize);
     }
 
     @Override
     public Result run() {
         try {
-            mailRepository.removeAll();
+            mailRepositories.forEach(Throwing.consumer(MailRepository::removeAll).sneakyThrow());
             return Result.COMPLETED;
-        } catch (MessagingException e) {
+        } catch (RuntimeException e) {
```

Comment:
Sneaky throws thows real MessagingException that you still need to catch here (or stop sneaky throwing ;-) )

Reply:
Since the sneakyThrow hide the `MessageException` from the compiler, I can't try to catch it :/

But I can't give a method reference throwing thing, so I'm trapped.